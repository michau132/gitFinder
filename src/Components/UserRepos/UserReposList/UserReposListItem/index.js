/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './style.css';
import Button from '../../../Button';

const UserReposListItem = ({
  listItem,
  selectUserRepo,
  hideSingleRepo,
  openSingleRepo,
}) => (
  <li className={classnames(styles.listItem, { [styles.active]: listItem.isChecked })}>
    <div className={styles.listItemInfo} onClick={() => selectUserRepo(listItem.id)}>
      <div className={styles.infoWrapper}>
        <input type="checkbox" checked={listItem.isChecked || false} readOnly />
        <h4 className={styles.infoName}>
          {listItem.name}
          (forks:
          {' '}
          {listItem.forks_count}
          ,
          stars:
          {' '}
          {listItem.stargazers_count}
          )
        </h4>
      </div>
      <p className={styles.infoDesc}>{listItem.description}</p>
    </div>
    <div className={styles.btnWrapper}>
      <Button
        type="button"
        onClick={() => openSingleRepo(listItem.html_url)}
        text="open"
      />
      <Button
        type="button"
        onClick={() => hideSingleRepo(listItem.id)}
        text="hide"
      />
    </div>
  </li>
);

UserReposListItem.defaultProps = {
  listItem: {
    name: null,
    id: null,
    html_url: null,
  },
};

UserReposListItem.propTypes = {
  listItem: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    forks_count: PropTypes.number,
    stargazers_count: PropTypes.number,
    description: PropTypes.string,
    html_url: PropTypes.string.isRequired,
  }),
  selectUserRepo: PropTypes.func.isRequired,
  hideSingleRepo: PropTypes.func.isRequired,
  openSingleRepo: PropTypes.func.isRequired,
};

export default UserReposListItem;
