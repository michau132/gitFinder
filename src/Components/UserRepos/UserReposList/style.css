fieldset {
  padding: 0;
}

.userRepos {
  overflow: hidden;
  max-height: inherit;
  min-width: 75%;
  max-width: 76%;
}

.listRepos {
  overflow: hidden;
  overflow-y: scroll;
  position: relative;
  height: 500px;
  list-style: none;
}