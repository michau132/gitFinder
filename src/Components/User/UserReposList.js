/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { List, Typography } from '@material-ui/core';
import styled from 'styled-components';
import UserReposListItem from './UserReposListItem';

const StyledFieldset = styled.fieldset`
  display: block;
  width: 100%;
`;

const StyledLegend = styled.legend`
  margin-left: 10px;
`;

const StyledList = styled(List)`
  list-style: none;
  height: 450px;
  overflow: hidden;
  overflow-y: scroll;
`;

const UserReposList = ({
  repos,
  ...restProps
}) => (
  <StyledFieldset>
    <StyledLegend>
      <Typography variant="subtitle1">
        Projects
      </Typography>
    </StyledLegend>

    <StyledList>
      {
        repos.map(listItem => (
          <UserReposListItem
            listItem={listItem}
            key={listItem.id}
            {...restProps}
          />
        ))
      }
    </StyledList>
  </StyledFieldset>
);

UserReposList.propTypes = {
  repos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      forks_count: PropTypes.number,
      stargazers_count: PropTypes.number,
      description: PropTypes.string,
      html_url: PropTypes.string.isRequired,
    }),
  ).isRequired,
  handleHideSingleRepo: PropTypes.func.isRequired,
  handleSelectUserRepo: PropTypes.func.isRequired,
};

export default UserReposList;
