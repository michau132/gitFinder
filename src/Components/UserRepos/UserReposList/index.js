import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';
import UserReposListItem from './UserReposListItem';

const UserReposList = ({
  userRepos,
  selectUserRepo,
  hideSingleRepo,
  openSingleRepo,
}) => (
  <fieldset>
    <legend>Projects</legend>
    <ul className={styles.listRepos}>
      {
        userRepos.map(listItem => (
          <UserReposListItem
            listItem={listItem}
            key={listItem.id}
            selectUserRepo={selectUserRepo}
            hideSingleRepo={hideSingleRepo}
            openSingleRepo={openSingleRepo}
          />
        ))
      }
    </ul>
  </fieldset>
);

UserReposList.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      forks_count: PropTypes.number,
      stargazers_count: PropTypes.number,
      description: PropTypes.string,
      html_url: PropTypes.string.isRequired,
    }),
  ).isRequired,
  hideSingleRepo: PropTypes.func.isRequired,
  openSingleRepo: PropTypes.func.isRequired,
  selectUserRepo: PropTypes.func.isRequired,
};

export default UserReposList;
