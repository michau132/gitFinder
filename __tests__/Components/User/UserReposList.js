import React from 'react';
import { shallow } from 'enzyme';
import { observable } from 'mobx';
import toJson from 'enzyme-to-json';
import UserReposList from '../../../src/Components/User/UserReposList';

const store = observable({
  repos: [
    {
      id: 1,
      name: 'firstRepo',
      forks_count: 0,
      stargazers_count: 0,
      description: 'First repo description',
      html_url: 'https://github.com/aOko123/firstRepo',
    },
    {
      id: 2,
      name: 'secondRepo',
      forks_count: 0,
      stargazers_count: 0,
      description: 'Second repo description',
      html_url: 'https://github.com/aOko123/secondRepo',
    },
  ],
});

describe('testing UserReposList component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      store,
    };

    wrapper = shallow(<UserReposList.wrappedComponent {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
