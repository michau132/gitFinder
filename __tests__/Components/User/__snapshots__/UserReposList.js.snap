// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserReposList component matching snapshot 1`] = `
<styled.fieldset>
  <styled.legend>
    <WithStyles(Typography)
      variant="subtitle1"
    >
      Projects
    </WithStyles(Typography)>
  </styled.legend>
  <Styled(WithStyles(List))>
    <UserReposListItem
      handleHideSingleRepo={[MockFunction]}
      handleSelectUserRepo={[MockFunction]}
      key="1"
      listItem={
        Object {
          "description": "First repo description",
          "forks_count": 0,
          "html_url": "https://github.com/aOko123/firstRepo",
          "id": 1,
          "name": "firstRepo",
          "stargazers_count": 0,
        }
      }
    />
    <UserReposListItem
      handleHideSingleRepo={[MockFunction]}
      handleSelectUserRepo={[MockFunction]}
      key="2"
      listItem={
        Object {
          "description": "Second repo description",
          "forks_count": 0,
          "html_url": "https://github.com/aOko123/secondRepo",
          "id": 2,
          "name": "secondRepo",
          "stargazers_count": 0,
        }
      }
    />
  </Styled(WithStyles(List))>
</styled.fieldset>
`;
